model/validation.go

```diff
@@ -669,3 +661,89 @@ func ValidateDestinationPolicy(msg proto.Message) error {
 
 	return errs
 }
+
+func validateProxyAddress(hostAddr string) error {
+	colon := strings.Index(hostAddr, ":")
+	if colon < 0 {
+		return fmt.Errorf("':' separator not found in %q, host address must be of the form <DNS name>:<port> or <IP>:<port>",
+			hostAddr)
+	}
+	port, err := strconv.Atoi(hostAddr[colon+1:])
+	if err != nil {
+		return err
+	}
+
+	if err = validatePort(port); err != nil {
+		return err
+	}
+	return nil
+}
+
+func validateDuration(pd *duration.Duration) error {
+	dur, err := ptypes.Duration(pd)
+	if err != nil {
+		return err
+	}
+	if dur <= 0 {
+		return errors.New("duration must be positive")
+	}
+	return nil
+}
+
+// ValidateProxyMeshConfig ...
+func ValidateProxyMeshConfig(mesh *proxyconfig.ProxyMeshConfig) (errs error) {
+	if mesh.EgressProxyAddress != "" {
+		if err := validateProxyAddress(mesh.EgressProxyAddress); err != nil {
+			errs = multierror.Append(errs, multierror.Prefix(err, "invalid egress proxy address:"))
+		}
+	}
+
+	// discovery address is mandatory since mutual TLS relies on CDS.
+	// strictly speaking, proxies can operate without RDS/CDS and with hot restarts
+	// but that requires additional test validation
+	if mesh.DiscoveryAddress == "" {
+		errs = multierror.Append(errs, errors.New("discovery address must be set to the proxy discovery service"))
+	} else if err := validateProxyAddress(mesh.DiscoveryAddress); err != nil {
+		errs = multierror.Append(errs, multierror.Prefix(err, "invalid discovery address:"))
+	}
+
+	if mesh.MixerAddress != "" {
+		if err := validateProxyAddress(mesh.MixerAddress); err != nil {
+			errs = multierror.Append(errs, multierror.Prefix(err, "invalid Mixer address:"))
+		}
+	}
+
+	if err := validatePort(int(mesh.ProxyListenPort)); err != nil {
+		errs = multierror.Append(errs, multierror.Prefix(err, "invalid proxy listen port:"))
+	}
+
+	if err := validatePort(int(mesh.ProxyAdminPort)); err != nil {
+		errs = multierror.Append(errs, multierror.Prefix(err, "invalid proxy admin port:"))
+	}
+
+	if mesh.IstioServiceCluster == "" {
+		errs = multierror.Append(errs, errors.New("Istio service cluster must be set"))
+	}
+
+	if err := validateDuration(mesh.DrainDuration); err != nil {
+		errs = multierror.Append(errs, multierror.Prefix(err, "invalid drain duration:"))
+	}
+
+	if err := validateDuration(mesh.ParentShutdownDuration); err != nil {
+		errs = multierror.Append(errs, multierror.Prefix(err, "invalid parent shutdown duration:"))
+	}
+
+	if err := validateDuration(mesh.DiscoveryRefreshDelay); err != nil {
+		errs = multierror.Append(errs, multierror.Prefix(err, "invalid refresh delay:"))
+	}
+
+	if err := validateDuration(mesh.ConnectTimeout); err != nil {
+		errs = multierror.Append(errs, multierror.Prefix(err, "invalid connect timeout:"))
+	}
+
+	if mesh.AuthCertsPath == "" {
+		errs = multierror.Append(errs, errors.New("invalid auth certificates path"))
+	}
```

Comment:
Can we also verify `mesh.AuthPolicy`? Then we can probably get rid of the check here: https://github.com/istio/manager/blob/master/proxy/envoy/discovery.go#L489